src/manifolds/Rotations.jl

```diff
@@ -390,8 +385,43 @@ function Base.show(io::IO, ::Rotations{n}) where {n}
     return print(io, "Rotations($(n))")
 end
 
+@doc raw"""
+riemannian_Hessian(M::Rotations, p, G, H, X)
+
+The Riemannian Hessian can be computed by adopting Eq. (5.6) [Nguyen:2023](@cite),
+so very similar to the Stiefel manifold.
+The only difference is, that here the tangent vectors are stored
+in the Lie algebra, u.e. the update direction is actually ``pX`` instead of jst ``X`` (in Stiefel).
+and that means the inverse has to be appliead to the (Euclidean) Hessian
+to map it into the Lie algebra.
+"""
+riemannian_Hessian(M::Rotations, p, G, H, X)
+function riemannian_Hessian!(::Rotations{N}, Y, p, G, H, X) where {N}
+    project!(SkewSymmetricMatrices(N), Y, p' * H - 1 / 2 .* X * (G' * p + p' * G))
```

Comment:
but then it would allocate? We have that in quite a few places actually, but I was unsure what is worse – allocation or multiplication. So we would write `Z = G'*p` in a line before? One could also just write a small `symm(X)` or use some linearly function because basically we symmetrize `G'*p`

Reply:
Ah, but I could check, I think all matrices are square anyways, so we can do `G'*p` in place of `Y` and not have an allocation anyways. Does that sound like a good idea?

Reply:
Using transposition does not cause additional allocations. `Z = G'*p` is exactly what is needed but we could indeed have a small function for symmetrization.

Reply:
But I noticed we can even use `Y` so we to store `G'*p` to avoid an allocation there?

Reply:
Yes, sure, I didn't notice your second comment when replying first. We can use `Y` to store `G'*p`, at least here and in unitary.

Reply:
Will check the others as well – on canonical Stiefel we might have an allocation instead for one of the interims things. I will check that and update the other ones as well :)